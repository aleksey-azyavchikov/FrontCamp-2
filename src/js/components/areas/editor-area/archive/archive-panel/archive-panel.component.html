<div>
    <div class="row">
        <div class="col">
            <span class="oi oi-plus icon"></span>
        </div>
    </div>
    <div cmIf="this.selected" class="row">
        <div class="col">
            <span class="oi oi-pencil icon"></span>
        </div>
    </div>
    <div cmIf="this.selected" class="row">
        <div class="col">
            <span class="oi oi-x icon"></span>
        </div>
    </div>
</div>