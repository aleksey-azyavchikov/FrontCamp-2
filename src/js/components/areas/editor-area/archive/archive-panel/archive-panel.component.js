import { Component } from "../../../../../core/decorators/component.decorator";
import BaseComponent from "../../../../base.component";
import { EditorMode } from "../../../../../core/enums/editor-mode.enum";
import { ActionType } from "../../../../../reducers/action-type";
import { CommandService } from "../../../../../services/command.service";
import { ApiInvoker } from "../../../../../core/api";
import { Endpoints } from "../../../../../core/endpoints";

@Component({
    selector: "fc-archive-panel",
    template: require("./archive-panel.component.html"),
    styles: require("./archive-panel.component.scss"),
    children: []
})
export default class ArchivePanelComponent extends BaseComponent {
    constructor() {
        super();
        this.subscriptions = [];
        this.selected = null;
        this.mode = null;
        this.EditorMode = EditorMode;
    }

    initializeHook() {
        this.selected = this.config.params.selected;
        this.mode = this.config.params.mode;

        let commandService = CommandService.getInstance();
        this.subscriptions.push(commandService.removeArticle
            .do(() => this.removeArticleHttp(this.selected._id))
            .subscribe()
        );

    }

    destroyHook() {
        this.subscriptions.forEach(subscription => subscription.unsubscribe());
    }

    defineDomElementsHook() {
        let domElements = {
            panel: this.config.ref.querySelector(".panel"),
            removeItemModal: $("#removeItemModal")
        };
        
        this.domElements = domElements;
    }

    bindHandlersHook() {
        this.bindEvent(this.domElements.panel, "click", (event) => { 
            this.processDeleteAction(event.target.id);
            this.dispatchMode(event.target.id);
        });
    }

    dispatchMode(mode) {
        this.config.store.dispatch({ 
            type: ActionType.SetArchiveEditorMode,
            payload: mode
        });
    }

    processDeleteAction(mode) {
        if(mode === EditorMode.Delete) {
            this.domElements.removeItemModal.modal("show");
        }
    }

    removeArticleHttp(id) {
        let commandService = CommandService.getInstance();
        let apiInvoker = ApiInvoker.getInstance();
        apiInvoker.invokeDelete(Endpoints.Articles({ id: id }))
            .then(() => commandService.updateArticles.next());
    }
}