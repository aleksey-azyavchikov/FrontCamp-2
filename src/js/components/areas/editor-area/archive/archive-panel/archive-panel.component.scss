.icon:hover {
    color: rgb(134, 217, 230);
    cursor: pointer;
}