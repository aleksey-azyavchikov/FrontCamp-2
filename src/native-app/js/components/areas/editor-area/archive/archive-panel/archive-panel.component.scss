.icon:hover, .selected-mode {
    color: rgb(134, 217, 230);
    cursor: pointer;
}

.panel div, div.panel {
    pointer-events: none
}

.panel span {
    pointer-events: auto
}
