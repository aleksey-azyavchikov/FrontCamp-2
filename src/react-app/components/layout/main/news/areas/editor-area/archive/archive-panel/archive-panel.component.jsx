

import React from "react";
import { connect } from "react-redux";
import { EditorMode } from "../../../../../../../../../core/enums/editor-mode.enum";
import { setEditorMode } from "../archive.actions";
import { OptionPanel } from "./option-panel/option-panel.component";
import "./archive-panel.component.scss";

class ArchivePanelPresenter extends React.Component {
    render() {
        const options = [{
            mode: EditorMode.Add,
            iconClass: "oi-plus"
        }, {
            mode: EditorMode.Edit,
            iconClass: "oi-pencil"
        }]

        const optionsComponent = options.map((option, index) => (
            <OptionPanel
                key={index}
                componentMode={option.mode}
                currentMode={this.props.mode}
                onModeClick={this.props.onModeClick}
                iconClass={option.iconClass}
            />
        ));

        return (
            <div class="panel">
                <ul class="list-inline">
                    {optionsComponent}
                </ul>
            </div>
        );
    }


}

export const ArchivePanel = connect(
    (state) => ({
        mode: state.archiveState.editorMode
    }),
    (dispatch) => ({
        onModeClick: (mode) => { dispatch(setEditorMode(mode)) }
    })
)(ArchivePanelPresenter)