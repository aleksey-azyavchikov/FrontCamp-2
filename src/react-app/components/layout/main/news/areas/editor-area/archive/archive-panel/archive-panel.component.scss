
.panel div, div.panel {
    pointer-events: none
}

.panel span {
    pointer-events: auto
}
