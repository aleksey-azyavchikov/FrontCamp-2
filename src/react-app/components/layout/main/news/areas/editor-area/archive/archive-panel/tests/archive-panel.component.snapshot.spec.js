import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { ArchivePanel } from "../archive-panel.component";
import { EditorMode } from "../../../../../../../../../../core/enums/editor-mode.enum";

describe("Compoent: OptionPanel", () => {
    const initialState = { archiveState: { editorMode: EditorMode.None } };
    const mockStore = configureStore();
    let store;
    let props;
    beforeAll(() => {
        store = mockStore(initialState);
        props = { store };
    })

    it("Display: initial", () => {
        const rendered = renderer.create(
            <ArchivePanel {...props}/>
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    })
}) 
