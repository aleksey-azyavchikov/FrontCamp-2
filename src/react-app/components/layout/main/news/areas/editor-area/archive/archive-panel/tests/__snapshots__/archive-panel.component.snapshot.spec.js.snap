// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compoent: OptionPanel Display: initial 1`] = `
<div
  className="panel"
>
  <ul
    className="list-inline"
  >
    <li
      className="list-inline-item"
    >
      <span
        className="panel-option oi oi-plus icon "
        id="Add"
        onClick={[Function]}
      />
    </li>
    <li
      className="list-inline-item"
    >
      <span
        className="panel-option oi oi-pencil icon "
        id="Edit"
        onClick={[Function]}
      />
    </li>
  </ul>
</div>
`;
